src/odemis/acq/align/test/roi_autofocus_test.py

```diff
@@ -0,0 +1,137 @@
+PURPOSE. See the GNU General Public License for more details.
+You should have received a copy of the GNU General Public License along with
+Odemis. If not, see http://www.gnu.org/licenses/.
+"""
+import logging
+import os
+import time
+import unittest
+from concurrent import futures
+
+import odemis
+from odemis import model
+from odemis.acq.align.autofocus import estimateAutoFocusTime
+from odemis.acq.align.roi_autofocus import autofocus_in_roi, estimate_autofocus_in_roi_time
+from odemis.util import testing
+
+logging.getLogger().setLevel(logging.DEBUG)
+logging.basicConfig(format="%(asctime)s  %(levelname)-7s %(module)s:%(lineno)d %(message)s")
+
+CONFIG_PATH = os.path.dirname(odemis.__file__) + "/../../install/linux/usr/share/odemis/"
+METEOR_CONFIG = CONFIG_PATH + "sim/meteor-sim.odm.yaml"
+
+
+class RoiAutofocusTestCase(unittest.TestCase):
+    """
+    Unit test for the roi_autofocus module
+    """
+
+    @classmethod
+    def setUpClass(cls):
+        testing.start_backend(METEOR_CONFIG)
+
+        # create some streams connected to the backend
+        # cls.microscope = model.getMicroscope()
+        cls.ccd = model.getComponent(role="ccd")
+        cls.focus = model.getComponent(role="focus")
+        cls.stage = model.getComponent(role="stage")
+
+        # Assumes the stage is referenced
+        cls.init_pos = (cls.stage.position.value["x"], cls.stage.position.value["y"])
+        cls.width, cls.height = cls.ccd.shape[:2]
+        cls.px_size = cls.ccd.getMetadata().get(model.MD_PIXEL_SIZE)
+
+        # Assume initial focus position is good
+        # The initial position comes from manual focussing the centre of overview image
+        current_focus = cls.focus.position.value["z"]  # assume manual focus before running the script
+        # Allow a range of +/- 30 µm around the focused position
+        cls.focus_range = (current_focus - 30.0e-6, current_focus + 30.0e-6)
+
+    @classmethod
+    def tearDownClass(cls):
+        logging.debug("moving back to initial position")
+        cls.stage.moveAbs({"x": cls.init_pos[0], "y": cls.init_pos[1]}).result()
+
+    def test_autofocus_in_roi(self):
+        """
+        Tests the autofus in roi is running and returns the focus points:
+        """
+        init_pos = self.init_pos
+        px_size = self.px_size
+        width = self.width
+        height = self.height
+        n_tiles = (3, 3)
+        overlap = 0.2
+
+        xmin = init_pos[0] - (1 - overlap) * n_tiles[0] / 2 * px_size[0] * width
+        ymin = init_pos[1] - (1 - overlap) * n_tiles[1] / 2 * px_size[1] * height
+        xmax = init_pos[0] + (1 - overlap) * n_tiles[0] / 2 * px_size[0] * width
+        ymax = init_pos[1] + (1 - overlap) * n_tiles[1] / 2 * px_size[1] * height
+
+        bbox = (xmin, ymin, xmax, ymax)
+
+        f = autofocus_in_roi(bbox, self.stage, self.ccd, self.focus, self.focus_range, n_tiles[0], n_tiles[1], 0)
+
+        # Test if the autofocus in roi is running
+        time.sleep(2)
```

Comment:
I am not sure how much time it takes. If I remove it, the test fails. Do I need to check proper time?

Reply:
Yes, you need a sleep() as immediately after calling it could have not yet started (as the thread needs a bit of time to start). However it should typically take less than 0.1s.

Reply:
sleep(0.1) works